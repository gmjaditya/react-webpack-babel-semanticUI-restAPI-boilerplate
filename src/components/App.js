import React,{Component} from 'react';
import {
  Grid,Segment,Pagination,Icon, Label, Menu, Table
}  from 'semantic-ui-react'

import HeaderColumn from './Header';
import SearchBox from './Search';
import apis from './api'

export default class  extends Component {
    state = {
      shipments:[]
    }
    componentWillMount(){
      apis.loadShipments().then(resp=>{
        this.setState({
          shipments:resp.data
        })
      })

    }
    render(){
      return(

        <Grid celled='internally' padded='horizontally' >
        <Grid.Row>
        <HeaderColumn/>
        </Grid.Row>
        <Grid.Row>
        <Grid.Column width={16}>
        <SearchBox/>
        </Grid.Column>
        </Grid.Row>
        <Grid.Row centered columns={2}>
        <Grid.Column >
        <Table celled>
          <Table.Header>
            <Table.Row>
              <Table.HeaderCell>Shipment Id</Table.HeaderCell>
              <Table.HeaderCell>Name</Table.HeaderCell>
              <Table.HeaderCell>Status</Table.HeaderCell>
              <Table.HeaderCell>Mode</Table.HeaderCell>
              <Table.HeaderCell>User</Table.HeaderCell>
            </Table.Row>
          </Table.Header>

          <Table.Body>
          {this.state.shipments.map(shipment=>{
            return(
              <Table.Row key={shipment.id}>
                <Table.Cell>{shipment.id}</Table.Cell>
                <Table.Cell>{shipment.name.split('from')[0]}<br/></Table.Cell>
                <Table.Cell>{shipment.status}</Table.Cell>
                <Table.Cell>{shipment.mode}</Table.Cell>
                <Table.Cell>{shipment.userId}</Table.Cell>
              </Table.Row>

            )
        })}


          </Table.Body>

          <Table.Footer>
            <Table.Row>
              <Table.HeaderCell colSpan='6'>
                <Menu floated='right' pagination>
                  <Menu.Item as='a' icon>
                    <Icon name='chevron left' />
                  </Menu.Item>
                  <Menu.Item as='a'>1</Menu.Item>
                  <Menu.Item as='a'>2</Menu.Item>
                  <Menu.Item as='a'>3</Menu.Item>
                  <Menu.Item as='a'>4</Menu.Item>
                  <Menu.Item as='a' icon>
                    <Icon name='chevron right' />
                  </Menu.Item>
                </Menu>
              </Table.HeaderCell>
            </Table.Row>
          </Table.Footer>
        </Table>
        </Grid.Column>

        </Grid.Row>
        </Grid>


      );
    }
  }
