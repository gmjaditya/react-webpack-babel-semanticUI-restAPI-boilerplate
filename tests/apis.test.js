import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from '../src/components/App';

import apis from '../src/api'
import db from '../db.json'
//snapshot test
describe('App component', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <BrowserRouter>
      <App/>
      </BrowserRouter>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// api test
describe('APIs Endpoint test', () => {
  it('should load all shipments', () => {
      apis.loadShipments()
      .then(resp=>{
        expect(resp.data).toEqual(db.shipments);
      })
      .catch(error=>console.log(error))

    });

  const index=Math.floor(Math.random() * Math.floor(db.shipments.length));
  it('should load a specific shipment', () => {
      apis.readShipment(db.shipments[index].id)
      .then(resp=>{
        expect(resp.data).toEqual(db.shipments[index]);
      })
      .catch(error=>console.log(error))

    });

  it('should update name of a specific shipment', () => {
      db.shipments[index].name='new name'
      apis.editShipment(db.shipments[index])
      .then(()=>{
        apis.readShipment(db.shipments[index].id)
        .then(resp=>{
          expect(resp.data).toEqual(db.shipments[index]);
        })
        .catch(error=>console.log(error))
      })
      .catch(error=>console.log(error))

    });
});
