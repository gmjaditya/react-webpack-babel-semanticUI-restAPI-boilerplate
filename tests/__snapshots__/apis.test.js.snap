// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot renders 1`] = `
<div
  className="ui celled vertically padded grid"
>
  <div
    className="row"
  >
    <div
      className="sixteen wide column"
    >
      <div
        className="ui header"
      >
        Freighthub Challenge
      </div>
    </div>
  </div>
  <div
    className="centered three column row"
  >
    <div
      className="three wide column"
    >
      <div
        className="ui basic label"
        onClick={[Function]}
      >
        Search Shipments by Id
      </div>
      <br />
      <br />
      <div
        className="ui search"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
      >
        <div
          className="ui icon input"
        >
          <input
            autoComplete="off"
            className="prompt"
            onChange={[Function]}
            onClick={[Function]}
            tabIndex="0"
            type="text"
            value=""
          />
          <i
            aria-hidden="true"
            className="search icon"
            onClick={[Function]}
          />
        </div>
        <div
          className="results transition"
        >
          <div
            className="message empty"
          >
            <div
              className="header"
            >
              No results found.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ten wide column"
    >
      <div
        className="ui segment"
      >
        <div
          className="ui large center aligned header"
        >
          All Shipments
        </div>
      </div>
      <table
        className="ui celled stackable striped table"
      >
        <thead
          className=""
        >
          <tr
            className=""
          >
            <th
              className=""
              onClick={[Function]}
            >
              Shipment Id  
              <i
                aria-hidden="true"
                className="sort icon"
                onClick={[Function]}
              />
            </th>
            <th
              className=""
              onClick={[Function]}
            >
              Name  
              <i
                aria-hidden="true"
                className="sort icon"
                onClick={[Function]}
              />
            </th>
            <th
              className=""
              onClick={[Function]}
            >
              Status  
              <i
                aria-hidden="true"
                className="sort icon"
                onClick={[Function]}
              />
            </th>
            <th
              className=""
              onClick={[Function]}
            >
              Mode  
              <i
                aria-hidden="true"
                className="sort icon"
                onClick={[Function]}
              />
            </th>
            <th
              className=""
              onClick={[Function]}
            >
              User  
              <i
                aria-hidden="true"
                className="sort icon"
                onClick={[Function]}
              />
            </th>
          </tr>
        </thead>
        <tbody
          className=""
        />
        <tfoot
          className=""
        >
          <tr
            className=""
          >
            <th
              className=""
              colSpan="6"
            >
              <div
                className="ui pagination right floated menu"
              >
                <a
                  className="icon item"
                  onClick={[Function]}
                >
                  <i
                    aria-hidden="true"
                    className="chevron left icon"
                    onClick={[Function]}
                  />
                </a>
                <a
                  className="icon item"
                  onClick={[Function]}
                >
                  <i
                    aria-hidden="true"
                    className="chevron right icon"
                    onClick={[Function]}
                  />
                </a>
              </div>
            </th>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>
`;
